pkg/ccl/changefeedccl/sink_cloudstorage.go

```diff
@@ -648,21 +662,62 @@ func (s *cloudStorageSink) flushFile(ctx context.Context, file *cloudStorageSink
 		return file.flushToStorage(ctx, s.es, dest, s.metrics)
 	}
 
-	s.flushGroup.Add(1)
-	go func() {
-		defer s.flushGroup.Done()
-		// NB: must use s.flushCtx; ctx may be short lived (i.e. cancelled).
-		if err := file.flushToStorage(s.flushCtx, s.es, dest, s.metrics); err != nil {
-			log.Errorf(ctx, "error flushing file to storage: %s", err)
-			// We must use the same type for error we store in flushErr.
-			s.flushErr.CompareAndSwap(nil, &flushError{error: err})
+	s.flushQueue.Lock()
+	defer s.flushQueue.Unlock()
+	if s.flushQueue.err != nil {
+		return s.flushQueue.err
+	}
+
+	s.flushQueue.q = append(s.flushQueue.q, destFile{file: file, dest: dest})
+	if len(s.flushQueue.q) == 1 {
```

Comment:
i'm going to rerun the tests to see if I can hit > 1 queued item.

Reply:
Still wasn't able to get that; basically, in my test, I was generating 1 file every 300ms or so, and it appear that most flushes took well under 100ms; I suppose it is possible for this queue to be non-zero; but that's fine -- as the PR says, if it becomes a problem, we can revisit.

Reply:
Why not just set the file size to like 1 byte in testing?

Reply:
Nah, i mean i wanted to see how this runs in production.   Of course we can do something like that in a unit test

Reply:
Verified with crazy small file size (4kb), files are queued (and then some): 
```
I221016 20:46:26.324904 164297 ccl/changefeedccl/sink_cloudstorage.go:671 ⋮ [n2,f‹eb3415c6›,job=805708385569374209] 1565589  changefeed flush queue depth 667347; ~2669388000 bytes to flush
```